lib/compressMeshes.js

```diff
@@ -0,0 +1,518 @@
+                mode : primitive.mode
+            };
+            var hashValue = hashObject(primitiveGeometry);
+            if (hashPrimitives[hashValue] !== undefined) {
+                // Copy compressed primitive.
+                copyCompressedExtensionToPrimitive(primitive, hashPrimitives[hashValue]);
+                return;
+            }
+            hashPrimitives[hashValue] = primitive;
+
+            var encoder = new encoderModule.Encoder();
+            var meshBuilder = new encoderModule.MeshBuilder();
+            var newMesh = new encoderModule.Mesh();
+
+            // First get the faces and add to geometry.
+            var indicesData = readAccessorPacked(gltf, gltf.accessors[primitive.indices]);
+
+            var indices = new Uint32Array(indicesData);
+            var numberOfFaces = indices.length / 3;
+            meshBuilder.AddFacesToMesh(newMesh, numberOfFaces, indices);
+
+            // Add attributes to mesh.
+            var attributeToId = {};
+            var attributes = primitive.attributes;
+            for (var semantic in attributes) {
+                if (attributes.hasOwnProperty(semantic)) {
+                    var attributeData = getNamedAttributeData(gltf, primitive, semantic);
+                    var numberOfPoints = attributeData.numberOfVertices;
+                    var attributeName = semantic;
+                    if (semantic.indexOf('_') !== -1) {
+                        attributeName = attributeName.substring(0, semantic.indexOf('_'));
+                    }
+                    var data = new Float32Array(attributeData.data);
+                    var attributeId = -1;
+                    if (attributeName === 'POSITION' || attributeName === 'NORMAL' ||
+                        attributeName === 'COLOR' ) {
+                        attributeId = meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule[attributeName],
+                            numberOfPoints, attributeData.numberOfComponents, data);
+                    } else if (semantic === 'TEXCOORD_0') {
+                        attributeId = meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule.TEX_COORD,
+                            numberOfPoints, attributeData.numberOfComponents, data);
+                    } else {
+                        attributeId = meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule.GENERIC,
+                            numberOfPoints, attributeData.numberOfComponents, data);
+                    }
+
+                    if (attributeId === -1) {
+                        throw new DeveloperError('Error: Failed adding attribute ' + semantic);
+                    } else {
+                        attributeToId[semantic] = attributeId;
+                    }
+                }
+            }
+
+            var encodedDracoDataArray = new encoderModule.DracoInt8Array();
+            encoder.SetSpeedOptions(10 - compressionLevel, 10 - compressionLevel);  // Compression level is 10 - speed.
```

Comment:
Somewhere we should validate that the input options are in correct ranges. Cesium has utility functions for this like: `Check.typeOf.number.greaterThan('width', width, 0);` It will throw an error if the input is out of range.

Reply:
Added some checks to compressDracoMeshes.